comp/remote-config/rcclient/rcclientimpl/rcclient.go

```diff
@@ -379,6 +387,11 @@ func (rc rcClient) agentTaskUpdateCallback(updates map[string]state.RawConfig, a
 		}(originalConfigPath, originalConfig)
 	}
 
+	err := streamlogs.StreamLogs(rc.Log, rc.Config, getDefaultStreamLogParams())
```

Comment:
Should this happen prior to us kicking off a goroutine, in the [previous block](https://github.com/DataDog/datadog-agent/blob/08e2671264a11a5e68a73ef651fa372e71f12ec1/comp/remote-config/rcclient/rcclientimpl/rcclient.go#L332-L388), that's going to produce the flare?

Your call to StreamLogs() happens afterwards, and I'm wondering if that opens up the possibility that not all of the log stream you want to capture is done before the flare tries to bundle it up. Maybe it should even be a part of the listener function that creates the flare? (cc @coignetp)